### ISSUE TITLE
Brush with autotransp act as in Color mode if same color is foreground

### ISSUE THREAD

**Issue by Sleuthgame** [mriale/PyDPainter#127]
**AutroTransp**

Defining brush with **AutoTransp** on.

If the 4 points of the square defining a new brush are all the same, this option ignore the current background color and define this colors as the new transparency color. Fine.

If, by coincidence, the background color defined for transparency is also the foreground color, a bug appears and the brush behaves as if in Color mode (uniformly this color) instead of Matte. Example with dark red 2.
![image](https://github.com/mriale/PyDPainter/assets/130657737/bb5aa428-bd0b-4945-9290-550060ada05c)

This also happens if you click on this color in the palette after defining the new brush.

**Comment by mriale** [comment]
I had to add a new `erase` flag to every drawing tool that draws brushes: line, freehand, circle, etc. This erase flag is set only when drawing with the right mouse button. Previously, the brush drawing function assumed that it should erase when the color passed in was that same as the background color. This didn't work right when both the foreground and background colors were the same, hence the bug. This has made the code more stable, but it did touch every drawing tool, so each has to be retested.

**Comment by Sleuthgame** [comment]
Fixed. 👍 

Tests :
It crash and exit with any solid background fill color using the **Area fill** tool.

**Comment by mriale** [comment]
Ooops! Found one I messed up! Fixed it.

**Comment by Sleuthgame** [comment]
Yes, everything working now.

**Comment by mriale** [comment]
Well actually, drawing tools that fill were doing the wrong thing too.

To recreate:
- right-click on the fill tool and set the `<->` fill mode
- click on color 20 in the palette (the first gray in the range)
- also right-click on color 20 so the foreground and background are the same
- choose the filled rectangle tool
- draw a rectangle with the left button
- draw a rectangle with the right button

Before the fix, both rectangles were a solid gray color.

After the fix, the rectangle drawn with the left button is a gradient and the one drawn with the right button is a solid gray color.

Again, I had to go though all the filled tools to fix this. I think I checked them all, but I could use another set of eyes.
